util.c

```diff
@@ -1932,6 +1932,58 @@ int mdmon_running(const char *devnm)
 	return 0;
 }
 
+/* wait_for_mdmon_control_socket() - Waits for mdmon control socket
+ * to be created within specified time.
+ * @devnm: Device for which mdmon control socket should start.
+ *
+ * In foreground mode, when mdadm is trying to connect to control
+ * socket it is possible that the mdmon has not created it yet.
+ * Give some time to mdmon to create socket. Timeout set to 10 sec.
+ */
+int wait_for_mdmon_control_socket(char *devname)
```

Comment:
after removing `subarray` determination, you can mark `devname` with const and possibly rename this to `const char *container_devnm`.

I'm trying to keep following approach for new code:
devname - means devnode path, like/dev/md127
devnm- means just device name md127

but either way it is fine for me, please try use const.